WorkloadManagementSystem/Utilities/QueueUtilities.py

```diff
@@ -0,0 +1,225 @@
+"""Utilities to help Computing Element Queues manipulation
+"""
+
+__RCSID__ = '$Id$'
+
+from DIRAC import S_OK, S_ERROR
+from DIRAC.Core.Utilities.List import fromChar
+from DIRAC.Core.Utilities.ClassAd.ClassAdLight import ClassAd
+from DIRAC.ConfigurationSystem.Client.Helpers.Resources import getDIRACPlatform
+from DIRAC.ConfigurationSystem.Client.Helpers.Operations import Operations
+from DIRAC.Core.Security.ProxyInfo import getProxyInfo
+
+
+def getQueuesResolved(siteDict):
+  """
+  Get the list of queue descriptions merging site/ce/queue parameters and adding some
+  derived parameters.
+
+  :param dict siteDict: dictionary with configuration data as returned by Resources.getQueues() method
+
+  :return: S_OK/S_ERROR, Value dictionary per queue with configuration data updated, e.g. for SiteDirector
+  """
+
+  queueDict = {}
+
+  for site in siteDict:
+    for ce in siteDict[site]:
+      ceDict = siteDict[site][ce]
+      qDict = ceDict.pop('Queues')
+      for queue in qDict:
+        queueName = '%s_%s' % (ce, queue)
+        queueDict[queueName] = qDict[queue]
+        queueDict[queueName] = qDict[queue]
+        queueDict[queueName]['Queue'] = queue
+        queueDict[queueName]['Site'] = site
+        # Evaluate the CPU limit of the queue according to the Glue convention
+        # To Do: should be a utility
+        if "maxCPUTime" in queueDict[queueName] and \
+           "SI00" in queueDict[queueName]:
+          maxCPUTime = float(queueDict[queueName]['maxCPUTime'])
+          # For some sites there are crazy values in the CS
+          maxCPUTime = max(maxCPUTime, 0)
+          maxCPUTime = min(maxCPUTime, 86400 * 12.5)
+          si00 = float(queueDict[queueName]['SI00'])
+          queueCPUTime = 60. / 250. * maxCPUTime * si00
```

Comment:
If future division is imported:
```suggestion
          queueCPUTime = 60 / 250 * maxCPUTime * si00
```

Reply:
If future division is imported, these two forms will give the same result I guess. So, is there any reason that the second form is better ?

Reply:
Yeah the future import makes them equivalent. I would favour removing the trailing periods as it's easier to read and better matches with what almost every Python package uses.